stl/inc/atomic

```diff
@@ -386,6 +386,19 @@ void _Atomic_wait_direct(
     }
 }
 
+template <class _Predicate>
+bool __stdcall _Atomic_wait_indirect_callback(const void* _Param, const void* _Comparand) {
+    auto* _Fn = static_cast<const _Predicate*>(_Param);

```

Comment:
Stylistic suggestion:
```suggestion
    const auto _Fn = static_cast<const _Predicate*>(_Param);
```
In general, we just use `auto`, even when we know the type is going to be a pointer (we can see it in the right hand side). And applying top-level `const` to local variables is nice.